charts/simple-app/templates/deployment.yaml

```diff
@@ -27,6 +27,9 @@ spec:
   template:
     metadata:
       annotations:
+        {{- if .Values.kms.secrets }}
+        checksum/config: {{ include (print $.Template.BasePath "/kmssecrets.yaml") . | sha256sum }}
+        {{- end}}
```

Comment:
I don't believe K8 will restart the pods when secrets are updated. (https://kubernetes.io/docs/concepts/configuration/secret/#environment-variables-are-not-updated-after-a-secret-update) 

I added this to maybe trigger a restart of the pods if secrets are modified. (referenced https://helm.sh/docs/howto/charts_tips_and_tricks/#automatically-roll-deployments) I'm exactly sure if this will work.

Reply:
This is plenty good. Nit, add a space on line 32 before the `}}`

Reply:
I simplified this a little further with `checksum/secrets: {{ toYaml .Values.kms.secrets | sha256sum }}`